workspace/notebook/manage_main_pipeline_config_entry.json

```diff
@@ -0,0 +1,373 @@
+					"\n",
+					"\n",
+					"new_entity_key =   # Run next cell to get the next available key\n",
+					"system_name = \"\"\n",
+					"object_name = \"\"\n",
+					"entity_name = \"\"\n",
+					"is_enabled = False  # Use the update cell at the bottom to enable it once function app is deployed using true"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"### Get Current Max Key"
+				]
+			},
+			{
+				"cell_type": "code",
+				"metadata": {
+					"collapsed": false
+				},
+				"source": [
+					"# Get current max key from main_pipeline_config\n",
+					"max_key_df = spark.sql(\"\"\"\n",
+					"    SELECT MAX(System_key) AS max_key\n",
+					"    FROM odw_config_db.main_pipeline_config\n",
+					"\"\"\")\n",
+					"\n",
+					"max_key = max_key_df.collect()[0][\"max_key\"] or 0\n",
+					"next_key = max_key + 1\n",
+					"\n",
+					"print(f\"Current max key: {max_key}\")\n",
+					"print(f\"Next available key: {next_key}\")\n",
+					""
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"### Insert New Entry"
+				]
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"# Safety checks before inserting\n",
+					"assert all([new_entity_key, system_name, object_name, entity_name]), \"Please fill in all parameters before inserting.\"\n",
+					"\n",
+					"# Check if entity already exists to prevent duplicates\n",
+					"exists = spark.sql(f\"\"\"\n",
+					"    SELECT COUNT(*) AS count\n",
+					"    FROM odw_config_db.main_pipeline_config\n",
+					"    WHERE entity_name = '{entity_name}'\n",
+					"\"\"\").collect()[0][\"count\"]\n",
+					"\n",
+					"if exists:\n",
+					"    raise ValueError(f\"Entity '{entity_name}' already exists in main_pipeline_config\")\n",
+					"\n",
+					"# Check if System_key already exists\n",
+					"key_exists = spark.sql(f\"\"\"\n",
+					"    SELECT COUNT(*) AS count\n",
+					"    FROM odw_config_db.main_pipeline_config\n",
+					"    WHERE System_key = {new_entity_key}\n",
+					"\"\"\").collect()[0][\"count\"]\n",
+					"\n",
+					"if key_exists:\n",
+					"    raise ValueError(f\"System_key {new_entity_key} already exists\")\n",
+					"\n",
+					"# Insert new entity into main_pipeline_config using the parameters\n",
+					"insert_sql = f\"\"\"\n",
+					"INSERT INTO odw_config_db.main_pipeline_config\n",
+					"(System_key, System_name, Object, entity_name, is_enabled)\n",
+					"VALUES ({new_entity_key}, '{system_name}', '{object_name}', '{entity_name}', {str(is_enabled).lower()})\n",
+					"\"\"\"\n",
+					"\n",
+					"spark.sql(insert_sql)\n",
+					"print(f\"Entity '{entity_name}' inserted successfully with System_key = {new_entity_key}\")\n",
+					""
```

Comment:
Just thinking, could a conditional SQL insert be done here instead? `df.count` is quite a slow operation. I'm not sure how big the table is, but if it's potentially large then I think it'll be worth doing. 
I think just moving the `WHERE` clause to be part of the `INSERT` statement should work

Reply:
Not a bad idea - I've done it like this cause the table is small with only 28 entities